### ISSUE TITLE
Expose Vite build.target option

### ISSUE THREAD

**Issue by steadygaze** [nuxt/nuxt#20065]
### Summary

I am filing this feature request to request that the [`build.target`](https://vitejs.dev/config/build-options.html#build-target) option from Vite be exposed. (I also suggest some other ideas that might prevent similar issues from being difficult to debug.)

### Story

I decided to try to use [top-level await](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/await#top_level_await) in some server code. This resulted in an error:

```
 ERROR  Error: Transform failed with 1 error:                                                                             nitro 21:07:17
/home/allen/projects/nuxitest/server/api/test.js:1:0: ERROR: Top-level await is not available in the configured target environment ("es2019")
```

Searching the error on Google brings up https://github.com/vitejs/vite/issues/6985 and [a relevant StackOverflow answer](https://stackoverflow.com/questions/72618944/get-error-to-build-my-project-in-vite-top-level-await-is-not-available-in-the). I tried adding

```js
// https://nuxt.com/docs/api/configuration/nuxt-config
export default defineNuxtConfig({
  // ...
  vite: {
    build: {
      target: "esnext"
    }
  }
  // ...
})
```

to my `nuxt.config.ts`, but this apparently had no effect, and I noticed the "configured target environment" from the error message was unchanged (`"es2019"`). I also tried `"es2021"` and adding a separate `vite.config.js` file, but these had no effect either.

After some frustration, I re-read the note

>>    Configuration that will be passed directly to Vite.
>
> See https://vitejs.dev/config for more information. Please note that not all vite options are supported in Nuxt.

in the [Nuxt config documentation](https://nuxt.com/docs/api/configuration/nuxt-config#vite) and it clicked that only the options specifically enumerated on the Nuxt side are supported/exposed.

### Implementation

I am not familiar with the codebase, but it looks like a 1-line change to `packages/schema/src/config/vite.ts` would expose this. (And also present the choice of what the default target should be, or maybe `null` would use the default from Vite?)

I would also suggest rephrasing the config documentation to not say "directly to", because that suggests that the `vite` field is passed directly to Vite like `vite.config.js`, like a passthrough (without necessarily being manipulated/culled). And also maybe say "Only the options explicitly listed here are exposed."

Not culling options from the `vite` field might also be nice to avoid having to do this for other options, but I'm not sure if there is some reason I don't know about that makes this a bad idea (maybe it would allow for some misconfigurations/bad interactions?).

I can file a separate bug for this, but I also noticed that `vite.build.target` actually autocompletes in VSCode and doesn't trigger any Typescript errors when run despite not being valid, which would suggest that Nuxt's config type definition just references the type definition from Vite. For example, just setting an undefined key in the config (if Typescript type checking is set up properly) results in

```
nuxt.config.ts:3:3 - error TS2345: Argument of type '{ bad: number; typescript: { strict: true; typeCheck: true; }; }' is not assignable to parameter of type 'NuxtConfig'.
  Object literal may only specify known properties, and 'bad' does not exist in type 'NuxtConfig'.

3   bad: 1,
    ~~~~~~
```

which is far more useful than silently accepting the config. I think sharing type definitions is probably a bad thing given that Nuxt's config type for `vite` is actually different/doesn't accept the same options. I tried to look into this as well but I don't understand Typescript well enough to trace through this.

## Related issues

When I checked existing issues, I noticed #14893, which would definitely be very nice to have/would have worked around my issue. I am trying to use this feature on the server side, so I don't want to change the target output for client side Javascript. However, as it stands, it doesn't appear to be possible to set the Vite target at all.

## Additional information

- [X] Would you be willing to help implement this feature?
- [ ] Could this feature be implemented as a module?

## Final checks

- [X] Read the [contribution guide](https://nuxt.com/docs/community/contribution).
- [X] Check existing [discussions](https://github.com/nuxt/nuxt/discussions) and [issues](https://github.com/nuxt/nuxt/issues).

**Comment by steadygaze** [comment]
After crawling through related issues, I stumbled upon `defineLazyEventHandler` mentioned in #14348 which does what I need. However, this could definitely be documented better; as it stands the only usage of it is the example [experimental/wasm](https://nuxt.com/docs/examples/experimental/wasm) and it doesn't appear in any of the regular documentation:

![image](https://user-images.githubusercontent.com/119086591/229700618-7fb67ad9-ed9a-47e5-a6c4-1412e86f76e5.png)

Basically, it shouldn't take this much digging to fix an issue like this. :sweat: :sob: I'd be willing to add docs for `defineLazyEventHandler` if regular contributors lack bandwidth and the relevant info isn't too hard for me to find/understand.

**Comment by danielroe** [comment]
The place where you want to use the await is actually not covered by vite but by nitro config, which has a separate target setting.

I think it should be possible to configure the vite target exactly as you did, but it just didn't appear to be working because it didn't apply to your server routes.
